plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -161,6 +164,68 @@ class="screen-reader-text"
 		return $block_content;
 	}
 
+	/**
+	 * Track feature usage of the Product Collection block within the site editor.
+	 *
+	 * @param int      $post_id  The post ID.
+	 * @param \WP_Post $post     The post object.
+	 *
+	 * @return void
+	 */
+	public function track_collection_instances( $post_id, $post ) {
+
+		if ( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST || ! wc_current_theme_is_fse_theme() ) {
+			return;
+		}
+
+		if ( ! $post instanceof \WP_Post ) {
+			return;
+		}
+
+		// Important: don't track revisions.
+		$post_type = $post->post_type;
+		if ( ! in_array( $post_type, array( 'post', 'page', 'wp_template', 'wp_template_part' ), true ) ) {
+			return;
+		}
+
+		if ( ! class_exists( 'WC_Tracks' ) || ! class_exists( 'WC_Site_Tracking' ) || ! \WC_Site_Tracking::is_tracking_enabled() ) {
+			return;
+		}
+
+		if ( ! has_block( $this->get_full_block_name(), $post ) ) {
+			return;
+		}
+
+		$blocks = parse_blocks( $post->post_content );
+		if ( empty( $blocks ) ) {
+			return;
+		}
+		// Count orders.
+		// Hint: Product count included in Track event. See WC_Tracks::get_blog_details().
+		$order_count = 0;
+		foreach ( wc_get_order_statuses() as $status_slug => $status_name ) {
+			$order_count += wc_orders_count( $status_slug );
+		}
```

Comment:
Just curious! Why are we counting orders here?
- Will it help us in tracking the size of the store(by order count), which is using the Product Collection block?

I just wanted to understand the reason behind it. Maybe there was discussion about it in the PR or somewhere else that I missed. :slightly_smiling_face:

Also, You mentioned `WC_Tracks::get_blog_details()` in the comment. Do you mean, similar to how it's tracking Product count, we are tracking order count as well?

Reply:
> Do you mean, similar to how it's tracking Product count, we are tracking order count as well?

That's correct, @imanish003! 

Each event requires the current product count (which is calculated within the WC_Tracks class) and the order count, which is added here. The code iterates the stock statuses because the `wc_orders_count` uses an internal cache.

Reply:
Got it. Thanks for the explanation! It makes sense now. 
Also, Thanks for also thinking about the cache while iterating the stock statuses. It's good to know that you have considered it. :slightly_smiling_face: